lib/shared/bucketing-assembly-script/assembly/bucketing/segmentation.ts

```diff
@@ -47,20 +47,28 @@ export function _evaluateOperator(
 }
 
 function doesUserPassFilter(
-    filter: AudienceFilterOrOperator, 
+    filter: AudienceFilterOrOperator,
     user: DVCPopulatedUser
 ): bool {
     if (filter.type === 'all') return true
     if (filter.type === 'optIn') return false
     if (!(filter instanceof UserFilter)) {
-        throw new Error('Invalid filter data')
+        console.log(`
+            [DevCycle] Error: Invalid filter data ${filter}.
+            To leverage this new filter definition, please update to the latest version of the DevCycle SDK in use
```

Comment:
not sure what `in use` means at the end here. Maybe just drop that?

Reply:
since Bucketing ASM can be used across different SDKs, but yeah it doesn't make much sense reading it back